### ISSUE TITLE
BUG HtmlEditorField insert link toggle broken with pre3.2 form ID/classes set

### ISSUE THREAD

**Issue by christopherdarling** [silverstripe/silverstripe-framework#7013]
On one of my sites we have the FormTemplateHelper_Pre32 enabled for forms, this seems to break the HtmlEditorField insert link feature, when selecting the link type the form options aren't updated. 

This looks to be down to https://github.com/silverstripe/silverstripe-framework/blob/3.6/javascript/HtmlEditorField.js#L582-L597 where it's searching for elements with an ID ending with _holder

When I disable the FormTemplateHelper_Pre32 extension, the insert link dialog works as expected.

**Comment by chillu** [comment]
Right, so this class: https://github.com/silverstripe/silverstripe-framework/blob/3/forms/FormTemplateHelper.php#L86. Maybe we need to force-remove this extension in `LeftAndMain->init()`?

**Comment by nglasl** [comment]
Adding my +1 onto this. I'm seeing issues in 3.6 with both the "add new page" form, and on the settings tab for the "parent ID" field. Neither of these are working correctly when the `FormTemplateHelper_Pre32` is enabled, since the JS targets `#Form_`, rather than `#CMSForm_`. This actually causes some nasty breakages, such as pages being forced back to the root of the site when saved.

**Comment by nglasl** [comment]
For anyone else that stumbles across this, I've put together the following that should correct any issues you're seeing (while this issue remains open).

<?php

/**
 *	This exists due to https://github.com/silverstripe/silverstripe-framework/issues/7013.
 */

class FormTemplateHelper36 extends FormTemplateHelper_Pre32 {

	public function isCMS() {

		return Controller::has_curr() && (Controller::curr() instanceof LeftAndMain);
	}

	public function generateFormID($form) {

		if($this->isCMS()) {
			return FormTemplateHelper::generateFormID($form);
		}
		else {
			return parent::generateFormID($form);
		}
	}

	public function generateFieldHolderID($field) {

		if($this->isCMS()) {
			return FormTemplateHelper::generateFieldHolderID($field);
		}
		else {
			return parent::generateFieldHolderID($field);
		}
	}

	public function generateFieldID($field) {

		if($this->isCMS()) {
			return FormTemplateHelper::generateFieldID($field);
		}
		else {
			return parent::generateFieldID($field);
		}
	}

}
